pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -189,12 +189,14 @@ func (b *Balancer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				if handler.name != cookie.Value {
 					continue
 				}
+
 				b.mutex.RLock()
 				_, ok := b.status[handler.name]
 				b.mutex.RUnlock()
 				if !ok {
-					continue
+					break
```

Comment:
```suggestion
					// because we already are in the only iteration that matches the cookie, so none
					// of the following iterations are going to be a match for the cookie anyway.
					break
```